src/cell_group.hpp

```diff
@@ -277,6 +258,14 @@ class cell_group {
     target_handle get_target_handle(cell_member_type target_id) const {
         return target_handles_[handle_partition_lookup(target_handle_divisions_, target_id)];
     }
+
+    void reset_samplers() {
+        // clear all pending sample events and reset to start at time 0
+        sample_events_.clear();
+        for(uint32_t i=0u; i<samplers_.size(); ++i) {
```

Comment:
There was actually a reason that a 32 bit unsigned type was used explicitly here. Using `std::size_t` instead and you get the following warning:
```
/users/bcumming/software/nestmc-proto/src/cell_group.hpp:266:32: warning: narrowing conversion of ‘i’ from ‘std::size_t {aka long unsigned int}’ to ‘uint32_t {aka unsigned int}’ inside { } [-Wnarrowing]
             sample_events_.push({i, sampler_start_times_[i]});
```
To keep the `size_t` index, we have to cast `i` to a narrower type inside the loop.
Which is implicitly saying "there are less than `2^32` samplers".

My fix is to have a `sample_event::size_type`, and explicitly use that for the index.